Client/game_sa/CModelInfoSA.cpp

```diff
@@ -1530,6 +1532,16 @@ void CModelInfoSA::RestoreOriginalModel()
         pGame->GetStreaming()->RemoveModel(m_dwModelID);
     }
 
+    // Restore original interface if model was converted
+    if (MapContains(m_convertedModelInterfaces, m_dwModelID))
+    {
+        CBaseModelInfoSAInterface* currentInterface = ppModelInfo[m_dwModelID];
+        ppModelInfo[m_dwModelID] = MapGet(m_convertedModelInterfaces, m_dwModelID);
```

Comment:
m_convertedModelInterfaces[m_dwModelID] can have invalid state here
Invalid loading state
Invalid pointer to rwObject
Wrong references counter
..

Reply:
1. How can m_convertedModelInterfaces[m_dwModelID] be invalid here?
2. The pointer to pRwObject is not relevant here because when this function is called, the game already calls DestroyRwObject and recreates it through the streamer

Reply:
Old references counter in m_convertedModelInterfaces can cause memory leak or random model unloading with a crash.
There is also the pColModel pointer that can reference to unloaded custom collision